kepcoApp/Db/kepco.db

```diff
@@ -29,20 +29,56 @@ record(stringin, "$(P)IDN")
     field(SDIS, "$(P)DISABLE")
 }
 
+record(ai, "$(P)FIRMWARE") {
+    field(DESC, "The firmware version of the KEPCO")
+    field(DTYP, "stream")
+    field(INP,  "@kepco.proto getFirmware $(PORT)")
+    field(PINI, "YES")
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:FIRMWARE")
+    field(SDIS, "$(P)DISABLE")
+    field(FLNK, "$(P)RSTRSND:ON_START.PROC")
+}
+
 record(stringin, "$(P)RST")
 {
-    field(DESC, "SCPI identification string")
+    field(DESC, "Reset the power supply")
     field(DTYP, "stream")
     field(INP,  "@kepco.proto RST $(PORT)")
-    field(PINI, "$(RESET=NO)")
     field(SIML, "$(P)SIM")
     field(SIOL, "$(P)SIM:RST")
     field(SDIS, "$(P)DISABLE")
 } 
 
+record(stringin, "$(P)RSTRSND") {
+    field(DESC, "Reset and resend setpoints to kepco")
+    field(DTYP, "stream")
+    field(INP, "@kepco.proto resetWaitResendSPs($(P)) $(PORT) ") # Reset and resend commands via proto so we can wait for reset to complete
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:RSTRSND")
+    field(SDIS, "$(P)DISABLE")
+}
+
+record(fanout, "$(P)RSND") {
+    field(LNK1, "$(P)CURRENT:SP.PROC")
+    field(LNK2, "$(P)VOLTAGE:SP.PROC")
+    field(LNK3, "$(P)OUTPUTSTATUS:SP.PROC")
+    field(LNK4, "$(P)OUTPUTMODE:SP.PROC")
+    field(SELM, "All")
+}
```

Comment:
I don't understand how this record is used as it's not referenced from GUI, ioc or as a forward link. I am also worried as this will process the records in a pseudo-random order which may give weird behaviour. If it's used can we chat about it, if not can it be removed?

Reply:
My mistake this was meant to be removed before (I was testing this rather than doing it through the proto), removed now